http2_general/test_h2_stream_priority.py

```diff
@@ -0,0 +1,758 @@
+        """
+        Try to open streams with the id range from 1 to 19, they should be reseted,
+        because of exceed of max concurent stream limit.
+        """
+        for i in range(0, client.h2_connection.remote_settings.max_concurrent_streams):
+            stream_id = client.stream_id
+            client.make_request(self.post_request)
+            self.assertTrue(client.wait_for_reset_stream(stream_id=stream_id))
+
+        return client, valid_req_num
+
+    def test_opening_created_idle_streams_after_exceed_max_concurent_streams_linit(self):
+        client, valid_req_num = self.test_max_concurent_stream_exceed_by_stream()
+
+        client.reinit_hpack_encoder()
+        client.valid_req_num = valid_req_num
+        # Opening of streams which was previously created with idle state is allowed.
+        for i in range(0, client.h2_connection.remote_settings.max_concurrent_streams):
+            client.make_request(self.post_request)
+
+        self.wait_for_responses(client, [21, 23, 25, 27, 29, 31, 33, 35, 37, 39])
+        self.check_response_sequence(client, 10, [21, 23, 25, 27, 29, 31, 33, 35, 37, 39])
+
+    def test_opening_not_created_idle_streams_after_exceed_max_concurent_streams_linit(self):
+        client, valid_req_num = self.test_max_concurent_stream_exceed_by_stream()
+
+        client.reinit_hpack_encoder()
+        client.valid_req_num = valid_req_num
+        client.stream_id = 41
+        client.make_request(self.post_request)
+        self.wait_for_responses(client)
+        self.check_response_sequence(client, 1)
+
+    def test_max_concurent_stream_exceed_by_priority_frame(self):
+        """
+        If creation of new stream leads to exceeding of max_concurrent_streams, we should reset
+        this stream. But according to RFC we can't reset idle streams, so Tempesta FW just
+        close the connetion with PROTOCOL_ERROR.
+        """
+        client, server = self.setup_test_priority(initial_window_size=1000)
+
+        self.assertTrue(client.h2_connection.remote_settings.max_concurrent_streams == 10)
+        for i in range(0, client.h2_connection.remote_settings.max_concurrent_streams - 1):
+            client.make_request(self.post_request)
+
+        client.send_bytes(
+            PriorityFrame(
+                stream_id=client.stream_id, depends_on=5, stream_weight=255, exclusive=False
+            ).serialize()
+        )
+        client.send_bytes(
+            PriorityFrame(
+                stream_id=client.stream_id + 2, depends_on=5, stream_weight=255, exclusive=False
+            ).serialize()
+        )
+
+        # Here we should wait for a long time since initial_window_size is small, and connection
+        # will be closed after all pending data will be send.
+        self.assertTrue(client.wait_for_connection_close(timeout=10))
+        self.assertIn(ErrorCodes.PROTOCOL_ERROR, client.error_codes)
+
+
+class TestBigHeadersAndBodyForSeveralStreams(TestPriorityBase, NetWorker):
```

Comment:
Why is this test needed here? I don't understand how it relates to prioritization/

Reply:
Yes you right, not needed.